packages/apollo-gateway/src/buildQueryPlan.ts

```diff
@@ -784,17 +872,25 @@ export class QueryPlanningContext {
     return isAbstractType(type) ? this.schema.getPossibleTypes(type) : [type];
   }
 
-  getVariableUsages(selectionSet: SelectionSetNode) {
+  getVariableUsages(selectionSet: SelectionSetNode, fragments: Set<FragmentDefinitionNode>) {
     const usages: {
       [name: string]: VariableDefinitionNode;
     } = Object.create(null);
 
+    const Variable: VisitFn<ASTNode, VariableNode> = (node) => {
+      usages[node.name.value] = this.variableDefinitions[node.name.value];
+    }
+
     visit(selectionSet, {
-      Variable: node => {
-        usages[node.name.value] = this.variableDefinitions[node.name.value];
-      },
+      Variable
     });
 
+    fragments.forEach(fragment => {
+      visit(fragment, {
```

Comment:
Would it make sense to pass a higher-level node to the `visit`or method and just have it traverse all of them in one pass or is that avoided for a particular performance reason?  Would seem to be a tad more comprehensive/defensive to do a single traversal for all `Variable` nodes, but it doesn't seem to be noted here why only selectionSet was passed in the first place.

Reply:
I'm actually unsure if there's a perf win here or not - I'd expect a similar result but only by speculation. I do think this tidies things up a bit anyhow, however. cc224d7